Access to XML IDs for individual Notes in a Chord.

boncata opened the pull request:
Added the property method 'noteIds'. It exposes the individual XML IDs
of Note objects inside a Chord object. These are the XML IDs given in a
MEI file.

This method was implemented by Werner Goebl and Martin Bonev on
the 7th February 2019.

Co-authored-by: Werner Goebl <wergo@users.noreply.github.com>

mscuthbert commented:
Your commit deletes the entire file and then rewrites it -- I suspect that you've changed the line encodings on the file or something like that.  Can you make a new push that only changes the relevant lines for reviewing?

Without seeing the changed lines, can you explain what it does that: `[n.id for n in ch]` would not do?  If it's not complex, it seems a bit of a niche method to include as a core music21 chord property.

boncata commented:
Hi, thank you for the quick response!

What our method does is exactly the same: return [n.id for n in self._notes], but we access the notes from the chord itself. 

In our project, we are reading MEI files and extracting the score information from them. What we need is the XML IDs of individual notes. To be honest, I was not aware that I could iterate over the notes inside a chord that way: [n.id for n in ch] (in hindsight, it is not a large leap of logic, given that the rest of the library functions in a similar way). That is why we added the functionality to the Chord class.

With the command you proposed we solve the problem we had and we are happy to use music21 that way.

 If you think the method we added can be useful for music21 keep it, but it only provides an alternative way to access individual note ids. In that sense, you can say it is niche. I did another push to my branch, adding only our method.

mscuthbert commented:
What we need is to make `.notes` return `._notes` cast to a tuple -- then it's more obvious how it works and how to do things like this.

boncata commented:
Yes, this would probably make it more obvious how to access individual notes in a chord.

mscuthbert commented:
`.notes` is added as a property in the current Git master and will be part of the next release.
